### PR TITLE
NeuralODE struct interface

### PR DESCRIPTION
**Pull request by jessebett**
Exporting a struct for `NeuralODE` that dispatches to `neural_ode` solve and whose `params` expose the underlying `dudt` to Flux makes the API nicer.

### CODE REVIEW

File: src/Flux/neural_de.jl [hunk 1]
```diff
@@ -1,10 +1,10 @@
-function neural_ode(model,x,tspan,
+function neural_ode(model,x,tspan, solver,
```
Comment (jessebett) [inline src/Flux/neural_de.jl]: I pulled `solver` out of `args` so it is an explicit field in the struct. I don't know if this is a good idea... @ChrisRackauckas 

Also what are some other examples of `args...` besides  the solver choice?

File: src/Flux/neural_de.jl [hunk 2]
```diff
@@ -24,3 +24,24 @@ function neural_dmsde(model,x,mp,tspan,
...
+struct NeuralODE
+    model
+    tspan
+    solver
+    args
+    kwargs
+end
+
+# Optional args and kwargs
+NeuralODE(model,tspan,solver,kwargs)= NeuralODE(model,tspan,solver,(),kwargs)
```
Comment (jessebett) [inline src/Flux/neural_de.jl]: This is related to the `args...` question. Is there a better way to allow both `args` and `kwargs` to be optional?

File: test/solver_options.jl [hunk 3]
```diff
@@ -0,0 +1,15 @@
+using OrdinaryDiffEq, StochasticDiffEq, Flux, DiffEqFlux, Test
+
+xs = Float32.(hcat([0.; 0.], [1.; 0.], [2.; 0.]))
+tspan = (0.0f0,25.0f0)
+dudt = Chain(Dense(2,50,tanh),Dense(50,2))
+
+@testset "only end" begin
```
Comment (jessebett) [inline test/solver_options.jl]: These tests need to be expanded to include other solver options. Especially the various ones which control the timepoints returned. I've noticed some combination of `save_*` break when trying to only return the final time step.

File: test/neural_de_gpu.jl [hunk 4]
```diff
@@ -146,3 +146,22 @@ end;
...
+    nn = Chain(
+               Dense(20,10,tanh),
+               Dense(10,10,tanh),
+               Dense(10,20,tanh)
+              ) |>gpu
+    fc = Chain(
+               Dense(20,10)
+              )|>gpu
+    nn_ode(x) = neural_ode(nn,x,gpu((0.f0,1.f0)), Tsit5(),save_everystep=false,reltol=1e-3,abstol=1e-3, save_start=false)
+    @test_broken typeof(nn(down(x))) == typeof(nn_ode(down(x)))
```
Comment (jessebett) [inline test/neural_de_gpu.jl]: I don't think this is still broken. This test needs to include a `allowscalar(false)` I will update this test tomorrow.

File: src/Flux/neural_de.jl (hunk 1 above)
Comment (ChrisRackauckas) [reply src/Flux/neural_de.jl]: Just use the solver one

File: src/Flux/neural_de.jl (hunk 2 above)
Comment (ChrisRackauckas) [inline src/Flux/neural_de.jl]:
```suggestion
NeuralODE(model,tspan,solver,args...;kwargs...)= NeuralODE(model,tspan,solver,args,kwargs)
```

Comment (ChrisRackauckas) [reply src/Flux/neural_de.jl]: The suggestion I did will naturally allow passing both and pull the splats into the appropriate fields

File: test/solver_options.jl (hunk 3 above)
Comment (ChrisRackauckas) [reply test/solver_options.jl]: interesting, yes please share

File: test/neural_de_gpu.jl (hunk 4 above)
Comment (ChrisRackauckas) [reply test/neural_de_gpu.jl]: where is `down(x)` from?

File: test/solver_options.jl (hunk 3 above)
Comment (ChrisRackauckas) [reply test/solver_options.jl]: these should all be fine nowadays.

**Comment by ChrisRackauckas** [comment]
Supersceded by https://github.com/JuliaDiffEq/DiffEqFlux.jl/pull/109

**Pull request closed without merging**